runtime/lua/vim/_editor.lua

```diff
@@ -307,14 +319,48 @@ end
 ---                            If a table, executes a single command. In this case, it is an alias
 ---                            to |nvim_cmd()| where `opts` is empty.
 ---@see |ex-cmd-index|
-function vim.cmd(command)
-  if type(command) == 'table' then
-    return vim.api.nvim_cmd(command, {})
-  else
-    return vim.api.nvim_exec(command, false)
-  end
+function vim.cmd(command) -- luacheck: no unused
+  error(command) -- Stub for gen_vimdoc.py
 end
 
+local VIM_CMD_ARG_MAX = 20
+
+vim.cmd = setmetatable({}, {
+  __call = function(_, command)
+    if type(command) == 'table' then
+      return vim.api.nvim_cmd(command, {})
+    else
+      return vim.api.nvim_exec(command, false)
+    end
+  end,
+  __index = function(t, command)
+    t[command] = function(...)
+      local opts
+      if select("#", ...) == 1 and type(select(1, ...)) == "table" then
```

Comment:
Maybe useful:  `vim.F.pack_len(...)`

Reply:
It wouldn't make this code any clearer and just introduces an unnecessary level of indirection.